src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = Object.assign({}, state);
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        currentState = Object.assign(currentState, action.extraData);
```

Comment:
```suggestion
        Object.assign(currentState, action.extraData);
```
Object.assign has already changed an object, there is no need to reassign the value for currentState

Reply:
it's a bit difficult, thank you!